mantidproject/mantid
# Move Muon HDF4 loaders to muon library

### Description of work
This PR moves the `LoadMuonNexus1` and `LoadMuonNexus2` algorithms to the `Framework/Muon` library instead of `Framework/DataHandling`. The base class for these two algorithms (`LoadMuonNexus`) is also moved. These two algorithms are the only algorithms in Mantid used to load HDF4 Nexus files. The plan is to isolate these two algorithms within the `Framework/Muon` library, and then convert the code to use the HDF4 API directly instead of the Nexus API (in a future PR).

Part of #38214 

### To test:
Check that the `LoadMuonNexus1` and `LoadMuonNexus2` algorithms are still accessible in the DataHandling section of the algorithms on Workbench

*This does not require release notes* because **its an internal change**

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
